### PR TITLE
[SPARK-48204][INFRA][FOLLOW] fix release scripts for the "finalize" step

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

Necessary fixes to finalize the spark 4.0 preview release. The major one is that pypi now requires API token instead of username/password for authentication.

### Why are the changes needed?

release

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

manual

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

File: dev/create-release/spark-rm/Dockerfile [hunk 1]
```diff
@@ -95,7 +95,7 @@ RUN curl -sS https://bootstrap.pypa.io/get-pip.py | pypy3
 RUN pypy3 -m pip install numpy 'six==1.16.0' 'pandas==2.2.2' scipy coverage matplotlib lxml
 
 
-ARG BASIC_PIP_PKGS="numpy pyarrow>=15.0.0 six==1.16.0 pandas==2.2.2 scipy plotly>=4.8 mlflow>=2.8.1 coverage matplotlib openpyxl memory-profiler>=0.61.0 scikit-learn>=1.3.2"
+ARG BASIC_PIP_PKGS="numpy pyarrow>=15.0.0 six==1.16.0 pandas==2.2.2 scipy plotly>=4.8 mlflow>=2.8.1 coverage matplotlib openpyxl memory-profiler>=0.61.0 scikit-learn>=1.3.2 twine==3.4.1"
```
Comment (cloud-fan) [inline dev/create-release/spark-rm/Dockerfile:98]: this is needed for pypi upload

**Comment by cloud-fan** [comment]
cc @HyukjinKwon @dongjoon-hyun

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by yaooqinn** [comment]
We need this for 3.5 and 3.4, can you send backport PRs @cloud-fan ?

**Comment by dongjoon-hyun** [comment]
+1 for backporting.

**Comment by cloud-fan** [comment]
backported to 3.5/3.4. I can't verify it now, we will verify it when we release the next 3.5/3.4 maintenance releases.

**Comment by dongjoon-hyun** [comment]
Thank you, @cloud-fan . Ya, sure.

**Pull request closed without merging**